pkg/cloud/vsphere/services/govmomi/service.go

```diff
@@ -120,7 +120,7 @@ func (vms *VMService) ReconcileVM(ctx *context.MachineContext) (infrav1.VirtualM
 func (vms *VMService) DestroyVM(ctx *context.MachineContext) (infrav1.VirtualMachine, error) {
 
 	vm := infrav1.VirtualMachine{
-		Name:  ctx.VSphereMachine.Name,
+		Name:  ctx.Machine.Name,
```

Comment:
Don't think this is forwards compatible. Need to re-evaluate _if_ we decide we want this change at all.

Reply:
I haven't been able to follow this conversation completely, but the name of an infra provider's machine is different than the CAPI provider's machine? I thought the names of `Cluster/InfraCluster` and `Machine/InfraMachine/KubeadmConfig` were always the same (for each group).

Reply:
btw, I was wrong earlier. This is forwards compatible cause we destroy VMs by `MachineRef` and not name.